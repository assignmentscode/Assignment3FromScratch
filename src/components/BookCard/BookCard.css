body, html {
    height: 100%;
    width: 100%;
    font-weight: 10;
    margin: 0px;
  }
.card {
    /* max-width: 280px; */
    flex-basis: 300px;
    width: fit-content;
    align-self: flex-start;
    text-align: center;
    font-family: arial;
    margin-top: 2%;
    margin-bottom: 2%;
    margin-left: 2%;
    margin-right: 2%;
    background-color: white;
    display: flex;
    flex-direction: column;
}
  .blog-image {
    width : 300px;
    height: 220px;
}
.post-heading {
    font-size: 20px;
    font-weight: 500;
    color: black;
    padding-left: 8px;
    padding-right: 8px;
    margin-bottom: 8px;
    text-align: center;
}
.heart-logo {
    float: right;
    padding: 3px 10px 10px 5px;
    width: 8%;
}
.ratings {
    float:left;
    padding-left: 5%;
    color: gray;
    font-weight: 700;
    font-size: 1.5em;
}