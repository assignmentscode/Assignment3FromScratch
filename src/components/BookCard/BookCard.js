import React from 'react';
import './BookCard.css';

const blackHeart = require('./../../Assets/heart-black.svg');
const redHeart = require('./../../Assets/heart-red.svg');
const imagePath = require('./../../Assets/cover_image.jpeg');

const BookCard = props => (
  <div className="card">
    <img src={imagePath} alt="Book" className="blog-image" />
    <p className="post-heading">{props.Name}</p>
    <div>
      <span className="ratings ">{props.rating}</span>
      <img src={(props.likedData === true) ? redHeart : blackHeart} alt="Heart" className="heart-logo" />
    </div>
  </div>
);
export default BookCard;
