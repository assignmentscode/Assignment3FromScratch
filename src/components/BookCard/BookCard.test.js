import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import BookCard from './BookCard';

describe('Button', () => {
  it('should render a Card describing Book', () => {
    const tree = renderer.create(<BookCard Name="Harry" rating="4.2" likedData />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should check the value passes to props', () => {
    const wrapper = shallow(<BookCard Name="Harry" rating="4.2" likedData />);
    // console.log(wrapper.props().children[1].props.children);
    expect(wrapper.props().children[1].props.children).toEqual('Harry');
  });
});
