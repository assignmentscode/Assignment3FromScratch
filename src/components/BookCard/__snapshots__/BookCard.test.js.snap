// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a Card describing Book 1`] = `
<div
  className="card"
>
  <img
    alt="Book"
    className="blog-image"
    src="cover_image.jpeg"
  />
  <p
    className="post-heading"
  >
    Harry
  </p>
  <div>
    <span
      className="ratings "
    >
      4.2
    </span>
    <img
      alt="Heart"
      className="heart-logo"
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "heart-red.svg",
        }
      }
    />
  </div>
</div>
`;
